frontend/src/pages/AccountSend.vue

```diff
@@ -590,20 +943,110 @@ function useSendForm() {
       });
       void txNotify(tx.hash, ethersProvider);
       await tx.wait();
-      resetForm();
+      await resetForm();
     } finally {
       isSending.value = false;
       showAdvancedSendWarning.value = false;
     }
   }
 
-  function resetForm() {
+  // Send funds
+  async function onBatchSendFormSubmit() {
+    try {
+      // Form validation
+      for (let i = 0; i < Sends.value.length; i++) {
+        if (!Sends.value[i].receiver || !Sends.value[i].token || !Sends.value[i].amount)
+          throw new Error('please complete the form');
+      }
+      if (!signer.value) throw new Error('wallet not connected');
+      if (!umbra.value) throw new Error('Umbra instance not configured');
+
+      // Verify the recipient ID is valid. (This throws if public keys could not be found. This check is also
+      // done in the Umbra class `send` method, but we do it here to throw before the user pays for a token approval.
+      // This should usually be caught by the isValidId rule anyway, but is here again as a safety check)
+      const ethersProvider = provider.value as Provider;
+      for (let i = 0; i < Sends.value.length; i++) {
+        await umbraUtils.lookupRecipient(Sends.value[i].receiver as string, ethersProvider);
+      }
+      // Ensure user has enough balance. We re-fetch token balances in case amounts changed since wallet was connected.
```

Comment:
Contrary to this comment I don't see a call that actually re-fetches token balances

Reply:
Uncommented the line that refetches balance https://github.com/ScopeLift/umbra-protocol/pull/491/commits/14f3447edb1b8fd70b273bdd1752da87e425c110